ts/components/LabelledItem/index.tsx

```diff
@@ -30,43 +36,123 @@ const styles = StyleSheet.create({
   },
   flex: {
     flex: 1
+  },
+  textInputMask: {
+    ...makeFontStyleObject("Regular")
   }
 });
 
+interface TextInputAdditionalProps extends TextInputProps {
+  disabled?: boolean;
+}
+
 type CommonProp = Readonly<{
-  label?: string;
+  accessibilityHint?: string;
+  accessibilityLabel?: string;
+  accessibilityLabelIcon?: string;
+  description?: string;
+  focusBorderColor?: string;
+  hasNavigationEvents?: boolean;
   icon?: string | ImageSourcePropType;
+  iconColor?: string;
   iconPosition?: "left" | "right";
   iconStyle?: StyleType;
-  iconColor?: string;
+  inputMaskProps?: TextInputMaskProps &
+    React.ComponentPropsWithRef<typeof TextInputMask>;
+  inputProps?: TextInputAdditionalProps;
   isValid?: boolean;
-  focusBorderColor?: string;
-  description?: string;
-  accessibilityLabel?: string;
-  accessibilityHint?: string;
-  accessibilityLabelIcon?: string;
+  label?: string;
   onPress?: () => void;
-  hasNavigationEvents?: boolean;
 }>;
 
-export type Props = WithTestID<CommonProp> & InputProps;
+export type Props = WithTestID<CommonProp>;
+
+function getColorsByProps({
```

Comment:
Do you find this refactoring suggestion an improvement?
from my POV it saves some ternary operator and it is more readable since it has less code

Note: I replaced `color(variables.brandGray).darken(0.2)` to its hex value `#BEC5CB`. In that way it is not computed on every render and its value can be reused
```typescript
type DescriptionColor = "bluegreyLight" | "bluegreyDark" | "red";
type LabelColor = Exclude<DescriptionColor, "red">;
type ColorByProps = {
  borderColor: string | undefined;
  descriptionColor: DescriptionColor;
  iconColor: string;
  labelColor: LabelColor;
  placeholderTextColor: string;
};
function getColorsByProps({
  isDisabledTextInput,
  hasFocus,
  isEmpty,
  isValid
}: {
  isDisabledTextInput: boolean;
  hasFocus: boolean;
  isEmpty: boolean;
  isValid?: boolean;
}): ColorByProps {
  if (isDisabledTextInput) {
    return {
      borderColor: IOColors.greyLight,
      descriptionColor: "bluegreyLight",
      iconColor: IOColors.bluegreyLight,
      labelColor: "bluegreyLight",
      placeholderTextColor: IOColors.bluegreyLight
    };
  }
  return {
    borderColor:
      hasFocus && isEmpty ? variables.itemBorderDefaultColor : undefined,
    descriptionColor: isValid === false ? "red" : "bluegreyDark",
    iconColor: variables.brandDarkGray,
    placeholderTextColor: variables.brandGrayDarken,
    labelColor: "bluegreyDark"
  };
}
```

Reply:
I like the refactoring, but `variables.brandGrayDarken` doesn't exist in [theme/variables](https://github.com/pagopa/io-app/blob/master/ts/theme/variables.ts).

Either I add it, or calculate it once per this module's lifespan like:

```typescript
const brandGrayDarken = color(variables.brandGray).darken(0.2)
```

What is your suggestion here?

Reply:
I don't know that these variables refer to style theme
👍🏻  for your code